File: backend/src/main/java/com/langleon/dsobuildsim/collectorbagbonus/dto/definition/CollectorBagCategoryBonusDefinitionDTO.java
package com.langleon.dsobuildsim.collectorbagbonus.dto.definition;

import com.langleon.dsobuildsim.collectorbagbonus.enums.CollectorBagCategory;

import java.util.List;


public record CollectorBagCategoryBonusDefinitionDTO(
        CollectorBagCategory category,
        String name,
        List<CollectorBagBonusDefinitionDTO> bonuses
) {
}

File: frontend/src/app/models/gamedataDTOs/CollectorBagCategoryBonusDefinitionDTO.ts
import { CollectorBagCategory } from "../../enums/CollectorBagCategory";
import { CollectorBagBonusDefinitionDTO } from "./CollectorBagBonusDefinitionDTO";

export interface CollectorBagCategoryBonusDefinitionDTO {
    category: CollectorBagCategory,
    name: string,
    bonuses: CollectorBagBonusDefinitionDTO[];
}